electrum/gui/qt/confirm_tx_dialog.py

```diff
@@ -70,18 +115,407 @@ def timer_actions(self):
+        elif self.feerate_e.isModified():
+            fee_color = ColorScheme.BLUE
+        else:
+            fee_color = ColorScheme.BLUE
+            feerate_color = ColorScheme.BLUE
+        self.fee_e.setStyleSheet(fee_color.as_stylesheet())
+        self.feerate_e.setStyleSheet(feerate_color.as_stylesheet())
+        #
+        self.needs_update = True
+
+    def update_fee_fields(self):
+        freeze_fee = self.is_send_fee_frozen()
+        freeze_feerate = self.is_send_feerate_frozen()
+        tx = self.tx
+        if self.no_dynfee_estimates and tx:
+            size = tx.estimated_size()
+            self.size_label.setAmount(size)
+            #self.size_e.setAmount(size)
+        if self.not_enough_funds or self.no_dynfee_estimates:
+            if not freeze_fee:
+                self.fee_e.setAmount(None)
+            if not freeze_feerate:
+                self.feerate_e.setAmount(None)
+            self.feerounding_icon.setIcon(QIcon())
+            return
+
+        assert tx is not None
+        size = tx.estimated_size()
+        fee = tx.get_fee()
+
+        #self.size_e.setAmount(size)
+        self.size_label.setAmount(size)
+        fiat_fee = self.main_window.format_fiat_and_units(fee)
+        self.fiat_fee_label.setAmount(fiat_fee)
+
+        # Displayed fee/fee_rate values are set according to user input.
+        # Due to rounding or dropping dust in CoinChooser,
+        # actual fees often differ somewhat.
+        if freeze_feerate or self.fee_slider.is_active():
+            displayed_feerate = self.feerate_e.get_amount()
+            if displayed_feerate is not None:
+                displayed_feerate = quantize_feerate(displayed_feerate)
+            elif self.fee_slider.is_active():
+                # fallback to actual fee
+                displayed_feerate = quantize_feerate(fee / size) if fee is not None else None
+                self.feerate_e.setAmount(displayed_feerate)
+            displayed_fee = round(displayed_feerate * size) if displayed_feerate is not None else None
+            self.fee_e.setAmount(displayed_fee)
+        else:
+            if freeze_fee:
+                displayed_fee = self.fee_e.get_amount()
+            else:
+                # fallback to actual fee if nothing is frozen
+                displayed_fee = fee
+                self.fee_e.setAmount(displayed_fee)
+            displayed_fee = displayed_fee if displayed_fee else 0
+            displayed_feerate = quantize_feerate(displayed_fee / size) if displayed_fee is not None else None
+            self.feerate_e.setAmount(displayed_feerate)
+
+        # set fee rounding icon to empty if there is no rounding
+        feerounding = (fee - displayed_fee) if (fee and displayed_fee is not None) else 0
+        self.set_feerounding_text(int(feerounding))
+        self.feerounding_icon.setToolTip(self.feerounding_text)
+        self.feerounding_icon.setIcon(read_QIcon('info.png') if abs(feerounding) >= 1 else QIcon())
```

Comment:
I decided to use an empty `QIcon()` instead of toggling visibility, because `setVisible()` affects the layout.
I pushed 6e8e8798c767299f93290d2c4a25c4eb284b0985 to address your issues